code/src/components/DetailSite.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router";
+import styled from "styled-components";
+import NotFound from "./NotFound";
+
+const BackLink = styled.div`
+  position: absolute;
+  display: inline-flex;
+  align-items: center;
+  left: 80px;
+  top: 50px;
+  opacity: 0.6;
+  transition: 0.3 ease;
+
+  a {
+    text-decoration: none;
+    font-weight: 700;
+    color: #fff;
+  }
+  &:hover {
+    opacity: 1;
+  }
+`;
+
+const DetailSite = () => {
+  const [details, setDetails] = useState({});
+  const [errorId, setErrorId] = useState(false);
+  const { id } = useParams();
+
+  useEffect(() => {
+    fetch(
+      `https://api.themoviedb.org/3/movie/${id}?api_key=5c4188456a9b734458844d02dc8962a9&language=en-US`
+    )
+      .then((res) => res.json())
+      .then((json) => {
+        if (json.id) {
+          setDetails(json);
+        } else {
+          setErrorId(true);
+        }
+      })
+      .catch(() => setErrorId(true));
+    // console.log("checking fetch", json);
+    //   setDetails(json);
+    // });
```

Comment:
Remember to delete console logs and out commented code (if you are not going to use it later on ofc)